RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc

```diff
@@ -0,0 +1,251 @@
+          bar.addVPSet("logWeightDenominatorByDetector", validator, vDefaults);
+        }
+        bar.add<double>("minAllowedNormalization", 1e-9);
+        pfClusterBuilder.add("allCellsPositionCalc", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<double>("corrTermLowE", 0.);
+        bar.add<double>("threshLowE", 6.);
+        bar.add<double>("noiseTerm", 21.86);
+        bar.add<double>("constantTermLowE", 4.24);
+        bar.add<double>("noiseTermLowE", 8.);
+        bar.add<double>("threshHighE", 15.);
+        bar.add<double>("constantTerm", 2.82);
+        pfClusterBuilder.add("timeResolutionCalcBarrel", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<double>("corrTermLowE", 0.);
+        bar.add<double>("threshLowE", 6.);
+        bar.add<double>("noiseTerm", 21.86);
+        bar.add<double>("constantTermLowE", 4.24);
+        bar.add<double>("noiseTermLowE", 8.);
+        bar.add<double>("threshHighE", 15.);
+        bar.add<double>("constantTerm", 2.82);
+        pfClusterBuilder.add("timeResolutionCalcEndcap", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        pfClusterBuilder.add("positionReCalc", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        pfClusterBuilder.add("energyCorrector", bar);
+      }
+      desc.add("pfClusterBuilder", pfClusterBuilder);
+    }
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  // the actual algorithm
+  std::unique_ptr<PFCPositionCalculatorBase> _positionCalc;
+  std::unique_ptr<PFCPositionCalculatorBase> _allCellsPositionCalc;
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  const edm::EDGetTokenT<reco::PFClusterHostCollection> alpakaPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitFractionHostCollection> alpakaPfRHFractionToken;
+  const edm::EDGetTokenT<reco::PFRecHitHostCollection> InputPFRecHitSoA_Token_;
+  const edm::ESGetToken<reco::PFClusterParamsHostCollection, JobConfigurationGPURecord> pfClusParamsToken;
+  const edm::EDPutTokenT<reco::PFClusterCollection> legacyPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitCollection> recHitsLabel;
+
+  int nRH = 0;
+};
+
+void LegacyPFClusterProducer::produce(edm::Event& event, const edm::EventSetup& setup) {
+  const reco::PFRecHitHostCollection& pfRecHits = event.get(InputPFRecHitSoA_Token_);
+
+  reco::PFClusterHostCollection const& clusterSoA = event.get(alpakaPfClustersToken);
+  reco::PFRecHitFractionHostCollection const& fractionSoA = event.get(alpakaPfRHFractionToken);
+
+  auto const& alpakaPfClusters = clusterSoA.const_view();
+  auto const& alpakaPfRhFrac = fractionSoA.const_view();
+
+  nRH = pfRecHits.view().size();
+  reco::PFClusterCollection out;
+  out.reserve(nRH);
+
+  auto const rechitsHandle = event.getHandle(recHitsLabel);
+
+  std::vector<int> seedlist1;
+  std::vector<int> seedlist2;
+  seedlist1.reserve(alpakaPfClusters.nSeeds());
```

Comment:
can you add a `reserve()` also for `seedlist2` ?

Reply:
Actually, it looks like `seedlist1` and `seedlist2` are filled and sorted, but never actually used ?
Can they be removed ?